instrumentation/apache-camel-2.20/src/test/groovy/test/TwoServicesCamelSpringBootBasedTest.groovy

```diff
@@ -0,0 +1,174 @@
+    setup:
+    createAndStartClient()
+    ProducerTemplate template = clientContext.createProducerTemplate()
+
+    when:
+    template.sendBody("direct:input", "Example request")
+
+    then:
+    assertTraces(1) {
+      trace(0, 9) {
+        it.span(0) {
+          name "input"
+          kind CLIENT
+          attributes {
+            "camel.uri" "direct://input"
+          }
+        }
+        it.span(1) {
+          name "input"
+          kind INTERNAL
+          attributes {
+            "camel.uri" "direct://input"
+          }
+        }
+        it.span(2) {
+          name "POST"
+          kind CLIENT
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "POST"
+            "$SemanticAttributes.HTTP_URL.key" "http://localhost:$portOne/serviceOne"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "camel.uri" "http://localhost:$portOne/serviceOne"
+          }
+        }
+        it.span(3) {
+          name "HTTP POST"
+          kind CLIENT
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "POST"
+            "$SemanticAttributes.HTTP_URL.key" "http://localhost:$portOne/serviceOne"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "$SemanticAttributes.NET_PEER_NAME.key" "localhost"
+            "$SemanticAttributes.NET_PEER_PORT.key" portOne
+            "$SemanticAttributes.NET_TRANSPORT.key" "IP.TCP"
+            "$SemanticAttributes.HTTP_FLAVOR.key" "1.1"
+          }
+        }
+        it.span(4) {
+          name "POST"
+          kind SERVER
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "POST"
+            "$SemanticAttributes.HTTP_URL.key" "http://localhost:$portOne/serviceOne"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "camel.uri" "http://0.0.0.0:$portOne/serviceOne"
+          }
+        }
+        it.span(5) {
+          name "POST"
+          kind CLIENT
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "POST"
+            "$SemanticAttributes.HTTP_URL.key" "http://0.0.0.0:$portTwo/serviceTwo"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "camel.uri" "http://0.0.0.0:$portTwo/serviceTwo"
+          }
+        }
+        it.span(6) {
+          name "HTTP POST"
+          kind CLIENT
+          attributes {
+            "$SemanticAttributes.HTTP_METHOD.key" "POST"
+            "$SemanticAttributes.HTTP_URL.key" "http://0.0.0.0:$portTwo/serviceTwo"
+            "$SemanticAttributes.HTTP_STATUS_CODE.key" 200
+            "$SemanticAttributes.NET_PEER_NAME.key" "0.0.0.0"
+            "$SemanticAttributes.NET_PEER_PORT.key" portTwo
+            "$SemanticAttributes.NET_TRANSPORT.key" "IP.TCP"
+            "$SemanticAttributes.HTTP_FLAVOR.key" "1.1"
+            "$SemanticAttributes.HTTP_USER_AGENT.key" "Jakarta Commons-HttpClient/3.1"
+          }
+        }
+        it.span(7) {
+          name "CachedChain.doFilter"
```

Comment:
Hmm, most HTTP server frameworks (JAX-RS, Spring, ...) usually rename their `SERVER` span so that it has the actual route as its name. IS it possible to do in camel?

Reply:
It's not Camel, it's a servlet instrumentation ;)))

Reply:
I know, bu the point stays: the "web framework" usually changes the server/servlet span name. Can it be done in camel?

Reply:
If I understand correctly - you're asking if Camel can set the name of the servlet that is being exposed (by Jetty component in this case) and instrumented by servlet instrumentation, currently generating span with "CachedChain.doFilter" name? IF this is the question the answer is - I don't think so (see https://camel.apache.org/components/latest/jetty-component.html). Besides it's not something that Camel instrumentation would do.

Reply:
Also discussed offline. Will propse something, although this seems tricky, due to Camel nature.

Reply:
Done for HTTP components, although I'm not a huge fan of the implementation.